packages/docusaurus-theme-classic/src/theme/AnnouncementBar/index.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, {useState} from 'react';
+import ExecutionEnvironment from '@docusaurus/ExecutionEnvironment';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import withForwardedRef from '@theme/hocs/withForwardedRef';
+
+import styles from './styles.module.css';
+
+const STORAGE_KEY = 'announcement_closed';
```

Comment:
Better to namespace it - `docusaurus.announcement.dismiss`.

I'd also recommend adding allowing a key for the announcement config. Otherwise users will never ever see new announcements after they've dismissed an announcement. Library authors can create new announcements but it will never show up for users who have dismissed announcements before.

We can allow the announcement config to take in a `key`/`id` prop that is added at the end of the storage key so that new announcements will still be visible to users who have dismissed older announcements - `docusaurus.announcement.dismiss.<id>`